dracut/repartition/endless-repartition.sh

```diff
@@ -147,18 +139,6 @@ if [ "$lastparttype" = "BFBFAFE7-A34F-448A-9A5B-6213EB736C22" \
   parts=$(echo "$parts" | sed '$d')
 fi
 
-# check the last partition on the disk
-lastpartno=$(echo "$parts" | sed -n -e '$ s/[^:]*\([0-9]\) :.*$/\1/p')
-
-if [ $lastpartno -eq $swap_partno ]; then
-  # already have an extra partition, perhaps we were halfway through creating
-  # a swap partition but didn't finish. Remove it to try again below.
```

Comment:
I don't think we should delete this block that removes any existing swap partition at the end of the disk. As the top of the script mentions:

> This script is written carefully so that it can be interrupted after each
> operation, on next boot it should effectively continue where it left off.

Another advantage of leaving this in place is that we can reclaim space used on an already-repartitioned system by writing a systemd unit that checks for the existence of a swap partition and re-adds the GUID:55 attribute on the root partition. On next boot, this unit will kick in from the initramfs and delete it. (We'd of course want to test this thoroughly…)

Reply:
In case it's not clear, the (unlikely) case I'm worried about is:

* User boots a new system with the old version of this script
* For some reason this script is interrupted after creating the swap partition but before unsetting the GUID:55 attribute
* A system update gets installed to bring in this new version of the script
* Now the new version of the script must cope with the presence of the swap partition